src/object_types/main_menu.lua

```diff
@@ -0,0 +1,328 @@
+                                              x + tw / 2 + th, y + th / 4)
+            end
+        end
+    end,
+
+    inputdown = function(this, key)
+        log.debug('Handling inputdown: %s', key)
+
+        if key == 'crouch' then
+            this.option = this.option + 1
+        end
+
+        if key == 'jump' then
+            this.option = this.option - 1
+        end
+
+        -- Handle video mode switching.
+        if this.state == this.STATE_OPTIONS and this.options_menu_option == 1 then
+            if key == 'left' or key == 'right' then
+                this.option_fullscreen = not this.option_fullscreen
+            end
+        end
+
+        -- Handle vsync switching.
+        if this.state == this.STATE_OPTIONS and this.options_menu_option == 2 then
+            if key == 'left' or key == 'right' then
+                this.option_vsync = not this.option_vsync
+            end
+        end
+
+        -- Handle msaa switching.
+        if this.state == this.STATE_OPTIONS and this.options_menu_option == 3 then
+            if key == 'left' then
+                this.option_msaa = math.max(this.option_msaa - 1, 0)
+            elseif key == 'right' then
+                this.option_msaa = math.min(this.option_msaa + 1, 4)
+            end
+        end
+
+        if key == 'ok' then
+            if this.state == this.STATE_MAIN then
+                -- Handle main menu buttons.
+
+                if this.main_menu_option == 1 then
+                    -- New game!
+                    -- Load the main map.
+                    map.load('test')
+                elseif this.main_menu_option == 2 then
+                    this.state = this.STATE_OPTIONS
+                    this.option = 1
+                    this:load_current_mode()
+                elseif this.main_menu_option == 3 then
+                    -- Quit game!
+                    love.event.quit(0)
+                end
+            elseif this.state == this.STATE_OPTIONS then
+                -- Handle options menu buttons.
+                -- Most options are selectors -- just handle confirm/cancel buttons.
+
+                if this.options_menu_option == 4 then
+                    -- Confirm button. Apply video settings and move to confirm menu.
+
+                    -- Grab the current mode and save it.
+                    this.last_w, this.last_h, this.last_flags = love.window.getMode()
+                    local w, h, flags = love.window.getMode()
+
+                    flags.msaa = this.option_msaa
+                    flags.vsync = this.option_vsync
+                    flags.fullscreen = this.option_fullscreen
+
+                    this:set_mode(w, h, flags)
+
+                    -- Switch state and set the confirm timer.
+                    this.state = this.STATE_OPTIONS_CONFIRM
+                    this.option = 1
+                    this.confirm_timer = this.CONFIRM_TIMER_LEN
```

Comment:
This confirm timer should be displayed on the screen to communicate to the player that the resolution will be switched back if nothing is entered.

Reply:
Added!